docs/onclusterlayering-quickstart.md

```diff
@@ -0,0 +1,486 @@
+# not actually be used.
+export imageRegistryPullspec="image-registry.openshift-image-registry.svc:5000/openshift-machine-config-operator/os-images:latest"
+
+
+yq -i e '.spec.buildInputs.baseImagePullSecret.name = strenv(baseImagePullSecretName)' ./layered-machineosconfig.yaml
+yq -i e '.spec.buildInputs.renderedImagePushSecret.name = strenv(pushSecretName)' ./layered-machineosconfig.yaml
+yq -i e '.spec.buildOutputs.currentImagePullSecret.name = strenv(pullSecretName)' ./layered-machineosconfig.yaml
+yq -i e '.spec.buildInputs.containerFile[0].content = strenv(containerfileContents)' ./layered-machineosconfig.yaml
+yq -i e '.spec.buildInputs.renderedImagePushspec = strenv(imageRegistryPullspec)' ./layered-machineosconfig.yaml
+```
+
+This yields the following YAML:
+
+```yaml
+---
+apiVersion: machineconfiguration.openshift.io/v1alpha1
+kind: MachineOSConfig
+metadata:
+  name: layered
+spec:
+  # Here is where you refer to the MachineConfigPool that you want your built
+  # image to be deployed to.
+  machineConfigPool:
+    name: layered
+  buildInputs:
+    containerFile:
+      # Here is where you can set the Containerfile for your MachineConfigPool.
+      # You'll note that you need to provide an architecture. This is because
+      # this will eventually support multiarch clusters.
+      - containerfileArch: noarch
+        content: |-
+          FROM quay.io/centos/centos:stream9 AS centos
+          RUN dnf install -y epel-release
+
+          FROM configs AS final
+          COPY --from=centos /etc/yum.repos.d /etc/yum.repos.d
+          COPY --from=centos /etc/pki/rpm-gpg/RPM-GPG-KEY-* /etc/pki/rpm-gpg/
+          RUN sed -i 's/\$stream/9-stream/g' /etc/yum.repos.d/centos*.repo && \
+              rpm-ostree install cowsay && \
+              ostree container commit
+    # Here is where you can select an image builder type. For now, we only
+    # support a pod image builder that we maintain ourselves. Future
+    # integrations can / will include other build system integrations.
+    imageBuilder:
+      imageBuilderType: PodImageBuilder
+    # The Machine OS Builder needs to know what pull secret it can use to pull
+    # the base OS image.
+    baseImagePullSecret:
+      name: global-pull-secret-copy
+    # Here is where you specify the name of the push secret you use to push
+    # your newly-built image to.
+    renderedImagePushSecret:
+      name: builder-dockercfg-123
+    # Here is where you specify the image registry to push your newly-built
+    # images to. In this example, we're using an ImageStream, but one can
+    # easily use an image registry such as Quay.io.
+    renderedImagePushspec: image-registry.openshift-image-registry.svc:5000/openshift-machine-config-operator/os-image:latest
+  buildOutputs:
+    # Here is where you specify what image will be used on all of your nodes to
+    # pull the newly-built image.
+    currentImagePullSecret:
+      name: builder-dockercfg-123
```

Comment:
Within this context, I think it can be left as-is because this represents the output of a script as opposed to a generic example where someone may be asked to fill in the blanks.